### DISCUSSION TITLE
How to handle long-running spans?

### DISCUSSION THREAD

**Discussion by Ayush-Kaushik** [open-telemetry/opentelemetry.io#5649] [Information Architecture]
We're looking for some advice on how to approach the integration of OpenTelemetry into a service that has long-running spans by nature, given issues that we've realized with respect to long-running spans (that they get auto-terminated if they remain outstanding for too long). 

We have a service which accepts incoming network connections from remote agent systems, where these connections remain active for potentially hours.  Occasionally a connection will be utilized to facilitate a request/response interaction to an agent, initiated by a third party that interacts with this service.  Our initial thinking was that the maintenance of a network connection by this service constituted a span, as there is ongoing activity (keep-alive messages are exchanged, even if nothing else occurs).  A request/response interaction would also be a span as well, linked to the connection span.
 
Unfortunately, as conceived above, we see that our connection spans are terminated because they persist for too long.  Given that a request/response interaction may at some point take up to 30s, it seems it is also in danger of being terminated based on what we've read here: [OpenTelemetry Challenges: Handling Long-Running Spans](https://thenewstack.io/opentelemetry-challenges-handling-long-running-spans/#:~:text=Here%E2%80%99s%20a%20few%20that%20people%20have%20used%3A%20Refactor,to%20rewrite%20span%20IDs%2C%20trace%20IDs%20and%20links.) about issues related to long-running spans.
 
Is there any general advice for handling services like this?  Should we not be thinking about spans at all as applying to these activities, or approaching the problem in another way?

**Comment by svrnm** [comment]
@Ayush-Kaushik the discussions on this repository are meant to talk about the information architecture of the OpenTelemetry website. 

On the topic there are discussions in the specification repository, you can take a look here for example: https://github.com/open-telemetry/opentelemetry-specification/issues/373
